record/record_test.go

```diff
@@ -332,22 +332,56 @@ func writeRecords(records [][]byte) ([]byte, error) {
 	return buf.Bytes(), nil
 }
 
-func TestBasicRecover(t *testing.T) {
-	records := [][]byte{
-		[]byte(strings.Repeat("a", blockSize-headerSize)),
-		[]byte(strings.Repeat("b", blockSize-headerSize)),
-		[]byte(strings.Repeat("c", blockSize-headerSize)),
+// generateTestRecords generates test records of specified lengths. The raw
+// records are returned as [][]byte and the serialized record form is returned
+// as a []byte. This allows the caller to compare the raw records with the
+// serialized versions. The first record will consist of repeating 0x00 bytes,
+// the next record of 0x01 bytes, and so forth. The values will loop back to
+// 0x00 after 0xff.
+func generateTestRecords(recordLengths ...int) ([][]byte, []byte, error) {
+	records := make([][]byte, len(recordLengths))
+	for i, n := range recordLengths {
+		records[i] = bytes.Repeat([]byte{byte(i)}, n)
 	}
+
 	buf, err := writeRecords(records)
+	return records, buf, err
+}
+
+func TestRecoverNoOp(t *testing.T) {
+	_, buf, err := generateTestRecords(blockSize-headerSize, blockSize-headerSize, blockSize-headerSize)
```

Comment:
I'd add line breaks to make it obvious how the three values compare:

_, buf, err := generateTestRecords(
  blockSize-headerSize,
  blockSize-headerSize,
  blockSize-headerSize,
)